lib/kvs/kvs_rparams.c

```
@@ -92,6 +95,44 @@ compression_value_algorithm_converter(
     return false;
 }
 
+static merr_t
+compression_value_algorithm_stringify(
+    const struct param_spec *const ps,
+    const void *const              value,
+    char *const                    buf,
+    const size_t                   buf_sz,
+    size_t *const                  needed_sz)
+{
+    int         n;
+    const char *param;
+
+    INVARIANT(ps);
+    INVARIANT(value);
+    INVARIANT(buf);
+
+    const enum vcomp_algorithm algo = *(enum vcomp_algorithm *)value;
+
+    switch (algo) {
+        case VCOMP_ALGO_NONE:
+            param = VCOMP_PARAM_NONE;
+            break;
+        case VCOMP_ALGO_LZ4:
+            param = VCOMP_PARAM_LZ4;
+            break;
+        default:
+            assert(false);
+    }
```

Default case should return an error...